pkg/logging/kafka/create.go

```diff
@@ -124,6 +136,42 @@ func (c *CreateCommand) createInput() (*fastly.CreateKafkaInput, error) {
 		input.Placement = fastly.String(c.Placement.Value)
 	}
 
+	if c.ParseLogKeyvals.Valid {
+		input.ParseLogKeyvals = fastly.CBool(c.ParseLogKeyvals.Value)
+	}
+
+	if c.RequestMaxBytes.Valid {
+		input.RequestMaxBytes = fastly.Uint(c.RequestMaxBytes.Value)
+	}
+
+	if c.UseSASL.Valid && c.UseSASL.Value {
+		if c.AuthMethod.Value == "" || c.User.Value == "" || c.Password.Value == "" {
+			return nil, fmt.Errorf("the --auth-method, --username,  and --password flags must be present when using the --use-sasl flag")
+
+		}
+
+		if c.AuthMethod.Value != "plain" &&
+			c.AuthMethod.Value != "scram-sha-256" &&
+			c.AuthMethod.Value != "scram-sha-512" {
+
+			return nil, fmt.Errorf("invalid SASL authentication method: %s. Valid values are plain, scram-sha-256, and scram-sha-512", c.AuthMethod.Value)
+		}
```

Comment:
This won't be required if we use the `EnumVar` arg type above, as the flag parser handles the validation for you.

Reply:
Done